src/sentry/discover/compare_tables.py

```diff
@@ -0,0 +1,193 @@
+import logging
+from enum import Enum
+from typing import TypedDict, cast
+
+import sentry_sdk
+
+from sentry.discover.arithmetic import is_equation
+from sentry.discover.dashboard_widget_split import _get_snuba_dataclass_for_dashboard_widget
+from sentry.discover.dataset_split import _get_equation_list, _get_field_list
+from sentry.discover.translation.mep_to_eap import (
+    QueryParts,
+    translate_columns,
+    translate_mep_to_eap,
+)
+from sentry.models.dashboard import Dashboard
+from sentry.models.dashboard_widget import DashboardWidget, DashboardWidgetQuery
+from sentry.models.organization import Organization
+from sentry.models.project import Project
+from sentry.search.eap.types import EAPResponse, SearchResolverConfig
+from sentry.search.events.types import SAMPLING_MODES, EventsResponse, SnubaParams
+from sentry.snuba import spans_rpc
+from sentry.snuba.metrics_performance import query as metrics_query
+
+logger = logging.getLogger(__name__)
+
+
+class CompareTableResult(Enum):
+    BOTH_FAILED = "both_requests_failed"
+    FIELD_NOT_FOUND = "field_not_found"
+    METRICS_FAILED = "metrics_failed"
+    NO_FIELDS = "no_fields"
+    NO_PROJECT = "no_project"
+    PASSED = "passed"
+    SNQL_EAP_FAILED = "snql_eap_failed"
+
+
+class CompareTableResultDict(TypedDict):
+    passed: bool
+    reason: CompareTableResult
+    fields: list[str] | None
+    widget_query: DashboardWidgetQuery
+    mismatches: list[str] | None
+
+
+def compare_table_results(metrics_query_result: EventsResponse, snql_eap_result: EAPResponse):
+    eap_data_row = snql_eap_result["data"][0] if len(snql_eap_result["data"]) > 0 else {}
+    metrics_data_row = (
+        metrics_query_result["data"][0] if len(metrics_query_result["data"]) > 0 else {}
+    )
+    mismatches = []
+    for field, data in metrics_data_row.items():
+        if is_equation(field):
+            continue
+        translated_field, *rest = translate_columns([field])
+        if data is not None and eap_data_row[translated_field] is None:
+            logger.info("Field %s not found in EAP response", field)
+            sentry_sdk.set_tag("dashboard_comparison_mismatch_field", field)
```

Comment:
i think this will just overwrite the `dashboard_comparison_mismatch_field` in each loop? so we'll only capture the last mismatched column.